crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1691,12 +1691,24 @@ impl Client {
     /// assert!(client.create_room(request).await.is_ok());
     /// # });
     /// ```
-    pub async fn create_room(&self, request: create_room::v3::Request) -> HttpResult<room::Joined> {
+    pub async fn create_room(&self, request: create_room::v3::Request) -> Result<room::Joined> {
+        let invite = request.invite.clone();
+        let is_direct_room = request.is_direct;
         let response = self.send(request, None).await?;
 
         let base_room =
             self.base_client().get_or_create_room(&response.room_id, RoomState::Joined).await;
-        Ok(room::Joined::new(self, base_room).unwrap())
+
+        let joined_room = room::Joined::new(self, base_room).unwrap();
+
+        // supporting just the DM chat with another person (no group dm chat)
+        if is_direct_room && invite.len() == 1 {
```

Comment:
Because I'm not sure how we are supposed to handle group chat.
If the user **A** creates a dm chat with **B** and **C** how would you updated **A**'s account data?

One option is like this:
```
{
  "content": {
    "B": [
      "room-id"
    ],
    "C": [
      "room-id"
    ]
  },
  "type": "m.direct"
}
```

But this also means that the A's client will think there is a dm chat **just** with B (or C).
This means that A's client cannot create a direct chat just with B (or C) anymore because we enforce just one DM chat for each pair of users.

Reply:
> Even if all Element clients didn't allow you add group DMs to m.direct, there's still tons of existing account data like this, and other clients will still be able to put group DMs in there as well.

I know, this doesn't work when other clients do different things, but at least we try hold it on Element.
What's the alternative for group chats anyway? Adding the room id just for the first member? Adding it for each member (like in the snippet in my previous comment)?

> For identifying an existing 1:1 room, you will have to iterate through known DM rooms with a user and look for one based on some heuristics (until we have canonical DMs).

Yeah possibly we want to rely on [this](https://github.com/matrix-org/matrix-rust-sdk/issues/1673) plus some check on the number of members (==2) maybe?

Reply:
> Yeah possibly we want to rely on [this](https://github.com/matrix-org/matrix-rust-sdk/issues/1673) plus some check on the number of members (==2) maybe?

I don't really know but that's not part of this PR anyways, right?

Reply:
Right

Reply:
As discussed privately: in group DMs we add the room N times in m.direct (once for each invited user).
https://github.com/matrix-org/matrix-rust-sdk/commit/5c0b2b53683cb249c55ee48dc740c363d20b70ef